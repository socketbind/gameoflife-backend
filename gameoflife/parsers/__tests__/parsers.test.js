const parsers = require('../');
const path = require('path');
const serializer = require('../../test/habitat_snapshot_serializer');

expect.addSnapshotSerializer(serializer);

const testDir = path.dirname(module.parent.filename);

test('can load life1.05 files', () => {
  const testFile = path.join(testDir, 'lif', 'max.lif');

  const simulation = parsers.lif.loadFile(testFile);
  return expect(simulation).resolves.toMatchSnapshot('loaded adder simulation');
});

test('fails gracefully when file does not exist', () => {
  const simulation = parsers.lif.loadFile('doesnotexist');
  return expect(simulation).rejects.toThrow(/ENOENT/);
});
