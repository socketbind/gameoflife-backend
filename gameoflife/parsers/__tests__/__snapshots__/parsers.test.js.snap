// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can load life1.05 files: loaded adder simulation 1`] = `
Simulation {
  "description": "MAX, the smallest known example of a \\"spacefiller\\"
This is the fastest-growing known pattern in Conway's Game of Life
(possibly the fastest possible). It fills space to a density of 1/2,
conjectured to be the maximum density, and does it at a speed of c/2
in each of the 4 directions, which has been proven to be the maximum
possible speed.
Population growth is [(t+19)^2+463]/4 for t divisible by 4;
.                    [(t+19)^2+487]/4 for t even, not div. by 4;
.                    [(t+18)^2+639]/4 for t odd.
Original construction, top/bottom stretchers by Hartmut Holzwart;
Size optimization, left/right stretchers by David Bell;
Original idea, middle part, l/r stretcher connection by Al Hensel.
This spacefiller by David Bell, September 1993.",
  "habitat": _____*_*_____________________
____*__*_____________________
___**________________________
__*__________________________
_****________________________
*____*_______________________
*__*_________________________
*__*_________________________
_*_________***___***_________
__****_*__*__*___*__*________
___*___*_____*___*___________
____*________*___*___________
____*_*______*___*___________
_____________________________
___***_____***___***_________
___**_______*_____*__________
___***______*******__________
___________*_______*_________
____*_*___***********________
___*__*__*____________**_____
___*_____************___*____
___*___*_____________*___*___
____*___************_____*___
_____**____________*__*__*___
________***********___*_*____
_________*_______*___________
__________*******______***___
__________*_____*_______**___
_________***___***_____***___
_____________________________
___________*___*______*_*____
___________*___*________*____
___________*___*_____*___*___
________*__*___*__*__*_****__
_________***___***_________*_
_________________________*__*
_________________________*__*
_______________________*____*
________________________****_
__________________________*__
________________________**___
_____________________*__*____
_____________________*_*_____
_____________________________
,
  "used_rule": [Function],
}
`;
